### PR TITLE
libinjector fix for Linux

### PR DESCRIPTION
**Pull request by skvl**
Fixes for several errors:
* `drakvuf_set_vcpu_regs` corrupts the state of Linux VM.
* Environment variables are not copied into `execve` syscall.

While experimenting with Linux I have experienced SIGSEGV error in host process after injection. By host I mean the process passed via `-i` option.

I have found that the trick in `handle_gprs_registers` function doesn't work for Linux:
handle_gprs_registers_fault

The image shows that after the `handle_gprs_registers` expected `RDI` and `RSI` registers are correct. But after `_int3_cb` their values are wrong. With VNC windows one could see the "Event Tracing" log collected with `echo 'syscalls:*' > /sys/kernel/debug/tracing/set_event`.

I have switched to usual `VMI_EVENT_RESPONSE_SET_REGISTERS` instead of `drakvuf_set_vcpu_regs`. And this fixed the issues.

After the fix I have found other issue. GUI applications failed to start without `DISPLAY` variable. This been fixed by coping inherited environment variables from `__environ` array after `vfork`.

### CODE REVIEW

**Comment by tklengyel** [comment]
Please don't leave code in there commented out.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
I haven't had a chance to do a final review on it yet, but hopefully in the next week or two we can close it.

**Comment by tklengyel** [comment]
Is the LibVMI submodule update intententional? Also, it looks like it's reverting to an older LibVMI commit?

File: src/libinjector/linux/linux_utils.h [hunk 1]
```diff
@@ -151,11 +151,12 @@ struct injector
     // KEEP THESE IN TOP and in sync with the order in injector_utils.c
     injector_step_t step;
     bool step_override; // set this as true for jumping to some arbitrary step
-    bool set_gprs_only;
+    bool dummy; // set_gprs_only
```
Comment (tklengyel) [inline src/libinjector/linux/linux_utils.h:154]: Doesn't seem to be used anywhere

**Comment by tklengyel** [comment]
Looks mostly OK, just these two minor things I noted above.

**Comment by skvl** [comment]
> Is the LibVMI submodule update intentional? Also, it looks like it's reverting to an older LibVMI commit?

I'm very sorry. This was by mistake.

File: src/libinjector/linux/linux_utils.h (hunk 1 above)
Comment (skvl) [reply src/libinjector/linux/linux_utils.h:154]: This is due to this comment:
image

This is used here https://github.com/tklengyel/drakvuf/blob/582ca8b1175a5aa56f640a0dbdb73b7d60b23732/src/libinjector/injector_utils.c#L116 and here https://github.com/tklengyel/drakvuf/blob/582ca8b1175a5aa56f640a0dbdb73b7d60b23732/src/libinjector/injector_utils.c#L142

**Comment by skvl** [comment]
I re-based the branch onto master

**Comment by skvl** [comment]
@tklengyel , the memdump memory error ( https://ci.drakvuf.com/job/DRAKVUF-windows7-sp1-x64/940/valgrindResult/pid=185276,0x0/ ) is not due to the branch.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Pull request merged**